src/test/groovy/com/bisnode/opa/client/rest/OpaRestClientSpec.groovy

```diff
@@ -0,0 +1,106 @@
+package com.bisnode.opa.client.rest
+
+import com.bisnode.opa.client.OpaClientException
+import com.bisnode.opa.client.OpaConfiguration
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.tomakehurst.wiremock.WireMockServer
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+
+import java.net.http.HttpClient
+import java.net.http.HttpResponse
+
+import static com.bisnode.opa.client.rest.ContentType.Values.APPLICATION_JSON
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import static com.github.tomakehurst.wiremock.client.WireMock.any
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+
+class OpaRestClientSpec extends Specification {
+
+    private static int PORT = 8181
+
+    private String url = "http://localhost:$PORT"
+    @Shared
+    private WireMockServer wireMockServer = new WireMockServer(PORT)
+    @Subject
+    private OpaRestClient client
+
+    def setupSpec() {
+        wireMockServer.start()
+    }
+
+    def cleanupSpec() {
+        wireMockServer.stop()
+    }
+
+    def setup() {
+        client = new OpaRestClient(new OpaConfiguration(url), HttpClient.newHttpClient(), new ObjectMapper())
+
+    }
+
+    def 'should remove trailing slashes from request URL to OPA'() {
+        given:
+          def path = '/v1/path/with/trailing/slash/'
+          wireMockServer
+                  .stubFor(any(anyUrl())
```

Comment:
I don't know if it's intentional, my first guess here would be to expect specific URL - without slash.

Reply:
The idea was for server to return always 200, so if there will be some exception thrown, it is not due to error message from server. Important part is one we will be veryfying in `then`